src/librustc_const_math/float.rs

```diff
@@ -203,3 +203,7 @@ impl ::std::ops::Neg for ConstFloat {
         ConstFloat { bits, ty: self.ty }
     }
 }
+
+/// This is `(f32::MAX + 0.5 ULP)` as an integer. Numbers greater or equal to this
+/// are rounded to infinity when converted to `f32`.
+pub const MAX_F32_PLUS_HALF_ULP: u128 = 0xffffff80000000000000000000000000_u128;
```

Comment:
Is there no way to obtain this from `rustc_apfloat`, using the max exponent or something?

Reply:
As discussed before, there are several ways but so far nobody found a way that's clearly better than hardcoding. I did discover this just now, which still isn't pretty or very intuitive, but tolerable:

```
pub const MAX_F32_PLUS_HALF_ULP: u128 =
  (1 << Single::PRECISION + 1) - 1 << (Single::MAX_EXP - Single::PRECISION as i16);
```

If you want, I'll update to use that expression instead of hardcoding.

Reply:
Where is that ` + 1` coming from? Is it supposed to be added to `Single::PRECISION`?